eng/common/pipelines/templates/steps/publish-1es-artifact.yml

```
@@ -23,7 +23,7 @@ steps:
     displayName: Set Failed Artifact Name
 
   - task: 1ES.PublishPipelineArtifact@1
-    condition: and(succeeded(), ${{ parameters.CustomCondition }})
+    condition: and(failedOrSucceeded(), ${{ parameters.CustomCondition }})
```

```suggestion
    condition: and(succeededOrFailed(), ${{ parameters.CustomCondition }})
```

[Ref](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#succeededorfailed)